packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-cognito-stack-builder.ts

```diff
@@ -103,13 +104,25 @@ export class AmplifyAuthCognitoStack extends cdk.Stack implements AmplifyAuthCog
     this.lambdaTriggerPermissions = {};
   }
   addCfnResource(props: cdk.CfnResourceProps, logicalId: string): void {
-    throw new Error('Method not implemented.');
+    if (!this._cfnResourceMap.has(logicalId)) {
+      this._cfnResourceMap.set(logicalId, new cdk.CfnResource(this, logicalId, props));
+    } else {
+      throw new Error(`Cfn Resource with LogicalId ${logicalId} doesnt exist`);
```

Comment:
I think the error message here should be:
```ts
    throw new Error(`Cfn Resource with LogicalId ${logicalId} already exists.`);
```

Similarly for the errors below

Reply:
Helo @jhockett, could you please review again. I think it is resolved now.
We really need that fix in amplify cli. Thank you!!